modules/openapi-generator/src/main/resources/php/composer.mustache

```diff
@@ -26,11 +26,18 @@
         "ext-curl": "*",
         "ext-json": "*",
         "ext-mbstring": "*",
-        "guzzlehttp/guzzle": "^6.2"
+        "php-http/discovery": "^1.14",
+        "psr/http-factory": "^1.0",
+        "php-http/client-common": "^2.3",
+        "guzzlehttp/psr7": "^1.8",
```

Comment:
Since this is still required, we haven't moved away from Guzzle. :thinking: IIUC, it's only used for the single method call, maybe we don't need it at all.

Reply:
`guzzlehttp/psr7` is a different from `guzzlehttp/guzzle`, which was the main target, so we got rid of the http client

But I think it is possible to use something else for `guzzlehttp/psr7`
We use 2 classes from it
```
use GuzzleHttp\Psr7\MultipartStream;
use GuzzleHttp\Psr7\Query;
```
Instead of `GuzzleHttp\Psr7\MultipartStream` we could use MultipartStreamBuilder from  https://github.com/php-http/multipart-stream-builder, also see https://docs.php-http.org/en/latest/components/multipart-stream-builder.html
But as far as I understand, there is no metapackage for that, we need to decide which PSR-7 MultipartStream builder we are going to use

As for `GuzzleHttp\Psr7\Query` I don't know if there is an alternative. It does something similar to what `http_build_query`, maybe https://packagist.org/packages/league/uri-query-parser?

Reply:
`guzzlehttp/psr7` provides the `psr/http-factory-implementation` so IMO we should at least look into trying to avoid using it directly, if possible. The reason is it doesn't allow us to swap out that implementation, so we might get the same issues as with the HTTP client, but from a different direction. :smile: 

We don't need to do it here, this is a great first step toward a HTTP agnostic package builder IMO. :+1:

Reply:
`guzzlehttp/psr7` is a bit confusing. It provides 
* PSR-7: HTTP message interfaces implementation
* PSR-17: HTTP Factories implementation
* and common utility methods 

We only use these common utility methods directly: `GuzzleHttp\Psr7\MultipartStream` and `GuzzleHttp\Psr7\Query`, they are not part of PSRs

Classes from `php-http/discovery` make the discovery process and it may or may not use guzzle's PSR implementation

We could discuss if we want to replace these utility methods as I suggested earlier, but I would say that it's a bit out of scope, because we don't use any PSR implementations directly

Reply:
guzzlehttp/psr7 v2 was released yesterday. https://github.com/guzzle/psr7/releases/tag/2.0.0